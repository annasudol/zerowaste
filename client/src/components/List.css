.fridge {
    display: block;
    width: 400px;
    height: 500px;
    background: url('../assets/images/svg/fridge.svg');
    background-repeat: no-repeat;
    background-size: 110% 110%;
    position: absolute;
    left: 48%;
    top: 50%;
    transform: translate(-50%, -50%);
    z-index: 0;
}

.fridge_inside {
  position: relative;
  top: 70px;
  z-index: 10;
  text-align: center;
}
