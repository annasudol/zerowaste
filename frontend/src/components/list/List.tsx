import React, { FC, ReactElement } from 'react';

import IconButton from '@material-ui/core/IconButton';
import DeleteIcon from '@material-ui/icons/Delete';

interface IngredientsListProps {
    list: string[]
    deleteItem?(item: string): void
}
export const List: FC<IngredientsListProps> = ({ list, deleteItem }): ReactElement => (
    <ol>
        {list?.map((item) => (
            <li key={item} className='ml-6'>
                <div className='w-full flex justify-between'>
                    <span className='mt-1'>{item}</span>
                    {deleteItem && <IconButton
                        aria-label='Delete'
                        onClick={() => deleteItem(item)}>
                        <DeleteIcon fontSize='small' />
                    </IconButton>}</div>
            </li>
        ))}
    </ol>
);
