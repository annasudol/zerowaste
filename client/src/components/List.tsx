import * as React from "react";
import Autocomplete from '@material-ui/lab/Autocomplete';
import { makeStyles } from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';
import { ButtonUI, Recipes } from '../components';

import products = require("../assets/data/products.json");
import { useQuery } from '@apollo/react-hooks';
import gql from 'graphql-tag';

const GET_RECIPES = gql`
  query GetRecipes($id: String) {
    recipeDetails(id: $id) {
      title
    }
  }
`;
const useStyles = makeStyles((theme) => ({
    root: {
        width: 500,
        '& > * + *': {
            marginTop: theme.spacing(3),
        }
    },
}));

export interface ListType {
    title: string
    id: number
}

export const List: React.FunctionComponent = () => {
    const [selectedProducts, setSelectedProducts] = React.useState<ListType[]>([{ title: "apple", id: 9003 }, { title: "sugar", id: 19335 }, { title: "flour", id: 20081 }]);
    const classes = useStyles();
    const changeProductList = (event: object, value: ListType[], reason: string) => {
        setSelectedProducts(value);
    }
    // tslint:disable-next-line: prefer-const
    // tslint:disable-next-line: no-shadowed-variable
    // const { loading, error, data } = useQuery(GET_RECIPES, {
    //     variables: { id: "591006" }
    // });
    // tslint:disable-next-line: no-console
    // console.log(data)


    // const { loading, error, data } = useQuery(GET_RECIPES, {
    //     variables: { id: "591006" }
    // });
    // tslint:disable-next-line: no-empty
    const searchRecipes = (): void => {
        // tslint:disable-next-line: no-shadowed-variable

    }


    return (<div>
        <div className={classes.root}>
            <Autocomplete
                multiple
                id="tags-outlined"
                options={products}
                getOptionLabel={option => option.title}
                defaultValue={[products[17], products[870], products[341]]}
                filterSelectedOptions
                onChange={(event, value, reason): void => changeProductList(event, value, reason)}
                includeInputInList={true}
                renderInput={params => (
                    <TextField
                        {...params}
                        variant="outlined"
                        label="Select grocery products"
                        placeholder="..."
                    />
                )}
            />
        </div>
        <ButtonUI size="medium" outlined={true} color="secondary" onClick={searchRecipes}>submit</ButtonUI>
        <Recipes recipesList={selectedProducts} />
    </div>)
}