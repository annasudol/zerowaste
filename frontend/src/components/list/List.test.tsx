import { shallow } from 'enzyme';
import React from 'react';
import IconButton from '@material-ui/core/IconButton';
import { List } from './List';
import toJson from 'enzyme-to-json';
const listProps = ["test1", "test2", "test3"];

const deleteItem = jest.fn();

describe('component', (): void => {
  describe('List', (): void => {
    it('match snapshot', (): void => {
      const wrapper = shallow(
        <List list={listProps} deleteItem={deleteItem}/>,
      );

      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders list items from props', (): void => {
        const wrapper = shallow(
          <List list={listProps} deleteItem={jest.fn()}/>,
        );

        wrapper.find("span").map((item, index): void => {
            expect(item.text()).toBe(listProps[index]);
        })
    });

    it('deleteItem function was called upon clicking delete icon', (): void => {
        const wrapper = shallow(<List list={listProps} deleteItem={()=>deleteItem()}/>);

        wrapper.find(IconButton).at(0).simulate('click');

        expect(deleteItem).toHaveBeenCalledTimes(1);

    })
  });
});
