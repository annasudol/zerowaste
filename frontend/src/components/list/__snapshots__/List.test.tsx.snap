// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component List match snapshot 1`] = `
<ol>
  <li
    className="ml-6"
    key="test1"
  >
    <div
      className="w-full flex justify-between"
    >
      <span
        className="mt-1"
      >
        test1
      </span>
      <WithStyles(ForwardRef(IconButton))
        aria-label="Delete"
        onClick={[Function]}
      >
        <DeleteIcon
          fontSize="small"
        />
      </WithStyles(ForwardRef(IconButton))>
    </div>
  </li>
  <li
    className="ml-6"
    key="test2"
  >
    <div
      className="w-full flex justify-between"
    >
      <span
        className="mt-1"
      >
        test2
      </span>
      <WithStyles(ForwardRef(IconButton))
        aria-label="Delete"
        onClick={[Function]}
      >
        <DeleteIcon
          fontSize="small"
        />
      </WithStyles(ForwardRef(IconButton))>
    </div>
  </li>
  <li
    className="ml-6"
    key="test3"
  >
    <div
      className="w-full flex justify-between"
    >
      <span
        className="mt-1"
      >
        test3
      </span>
      <WithStyles(ForwardRef(IconButton))
        aria-label="Delete"
        onClick={[Function]}
      >
        <DeleteIcon
          fontSize="small"
        />
      </WithStyles(ForwardRef(IconButton))>
    </div>
  </li>
</ol>
`;
